src/repositories.test.js

```diff
@@ -0,0 +1,31 @@
+const repositories = require('./repositories')
+const assert = require('assert')
+
+describe('repositories', () => {
+    it('should create an empty list', () => {
+        assert.deepEqual(repositories.all, new Map())
+    })
+
+    it('should throw an error when no id is provided and the repository has no name', () => {
+        assert.throws(() => repositories.add(class { }), /^Error: Herbarium requires an id for this repository or a non anonymous function\/class.$/)
+    })
+
+    it('should create an item with the provided id and repository', () => {
+        class Repository { }
+        const id = 'an id'
+
+        const addedRepository = repositories.add(Repository, id)
+
+        assert.equal(addedRepository.id, id)
+        assert.equal(addedRepository.repository, Repository)
+    })
+
+    it('should create an item with the repository name as id when no id is provided', () => {
+        class Repository { }
```

Comment:
will it work with herbs2knex and other repositories?

Reply:
Yes, but there would be necessary to add the dev-dependency of `@herbsjs/herbs2knex`.

Reply:
the issue here is broader. since repositories have no "description" ideally we should have tests for all official herbs repositories. Not all use the same class / function structures.

to be honest I don't even think we should support this convention for repositories now, for the lack of metadata.